bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java

```diff
@@ -307,9 +325,18 @@ public BigtableAsyncAdmin(BigtableAsyncConnection asyncConnection) throws IOExce
     throw new UnsupportedOperationException("restoreSnapshot"); // TODO
   }
 
+  /* (non-Javadoc)
+   * @see org.apache.hadoop.hbase.client.AsyncAdmin#truncateTable(org.apache.hadoop.hbase.TableName, boolean)
+   */
   @Override
-  public CompletableFuture<Void> truncateTable(TableName arg0, boolean arg1) {
-    throw new UnsupportedOperationException("truncateTable"); // TODO
+  public CompletableFuture<Void> truncateTable(TableName tableName, boolean preserveSplits) {
+  	return CompletableFuture.supplyAsync(() -> {
+      DropRowRangeRequest.Builder dropBuilder = DropRowRangeRequest.newBuilder().setDeleteAllDataFromTable(true);
```

Comment:
I think that you can just create a `DropRowRangeRequest` outside of `CompletableFuture`, and then just return:

```
FutureUtils.toCompletableFuture(bigtableTableAdminClient.dropRowRangeAsync(request))
.thenApply(r -> null);
```